edb/server/compiler/errormech.py

```diff
@@ -139,24 +155,29 @@ def interpret_backend_error(schema, fields):
     column_name = fields.get('c')
     constraint_name = fields.get('n')
 
-    if code == PGError.NotNullViolationError:
-        source_name = pointer_name = None
-
-        if schema_name and table_name:
-            tabname = (schema_name, table_name)
+    return ErrorDetails(
```

Comment:
I don't like functions that have polymorphic return types.  I this case, it's either `ErrorDetails` or an actual exception instance.  I suggest to have one function to build ErrorDetails, and another that acts on it.

Reply:
OK, I've added a handler for some of these error details.